lib/archive_importer/post_importer.rb

```diff
@@ -17,6 +17,7 @@ def substitute_author
 
       entity_data["photos"].each do |photo|
         photo["entity_data"]["author"] = user.diaspora_handle
+        photo["entity_data"]["remote_photo_path"] = "#{AppConfig.pod_uri}uploads\/images\/"
```

Comment:
This only changes the photos URL if the post doesn't exist on the pod yet, changing it for posts which were already federated to this pod before the migration need to be changed separately.

Also this only changes the URLs on the pod where the user migrates to, all other pods which have received the posts from the old pod will still load the photos from there. So photos also need a way to update existing posts (both on the target pod and also on other remote pods), which hopefully can be done the same way. Sending a new photo message for each photo isn't a solution because that would just fill all sidekiq queues on all pods with photo messages. 

Also it also can't just blindly be set to the url of the new pod, because if the new pod uses something like S3, then the `remote_photo_path` would be the path to S3.

Reply:
??? 
So.. how do a new photo URL can be send out to other pods to update their post?

Reply:
As discussed elsewhere - this will be done elsewhere. (Protocol, Federation?)

Reply:
The first part of the comment is still valid and belongs to this PR:

> This only changes the photos URL if the post doesn't exist on the pod yet, changing it for posts which were already federated to this pod before the migration need to be changed separately.

It only changes the URL of photos which aren't in the database already. And during testing yesterday I also noticed, that it does this during the import of the JSON, so it then imports the photos with a broken URL (until I also import a photos archive). So this shouldn't happen here at all, it should happen during the photo import, and that way you also fix the problem that it ignores photos which are already in the database.